dcpy/connectors/edm/recipes.py

```diff
@@ -26,6 +30,76 @@
 ValidSocrataFormats = Literal["csv", "geojson", "shapefile"]
 
 
+#### extract objects
+class RawDatasetKey(BaseModel, extra="forbid"):
+    name: str
+    timestamp: datetime
+
+    @property
+    def s3_path(self):
+        return Path(RAW_FOLDER) / self.name / self.timestamp.isoformat()
+
+
+class DatasetKey(BaseModel, extra="forbid"):
+    name: str
+    version: str
+
+
+class ExtractConfig(BaseModel, extra="forbid"):
```

Comment:
This is largely ported from dcpy/extract/config.py. If these types are going to be in config files in edm-recipes, wanted them here. Much like "libraryconfig" below. For now, differentiating them by dcpy module name

Reply:
This feels like part of a broader conversation, and doesn't have to be addressed as part of this, but I do wonder if there's a better spot for this. I suppose I've been conceiving of `edm.connectors` as a set of utilities to interact with collections of items on s3, without necessary having a knowledge about those items. `connectors.edm.publishing` being a good example. 

Curious if anything except for `dcpy.extract` use the Source classes? The Socrata orgs in particular feel misplaced.

Reply:
Yeah. We talked a bit about this in person obviously, but just to recap - these types by domain certainly belong more in dcpy.extract, while connectors.edm.recipes more just moves them around. But it's still nice for type safety for recipes to at least know what they are.

Would probably make sense to move out `models` of different domains to a separate module in dcpy, so that they can be organized more logically by domain but still available to code throughout dcpy without recipes connector and extract code being mutually dependent on one another